ChiTech/ChiPhysics/PhysicsMaterial/transportxsections/prop_transportxs_01b_readfromchi.cc

```diff
@@ -24,20 +24,161 @@ SIGMA_T_END
 
 Comments Bla Bla
 
+TRANSFER_MOMENTS_BEGIN
+#Zeroth moment (l=0)
+M_GPRIME_G_VAL 0 0 0 0.01
+M_GPRIME_G_VAL 0 0 1 0.09
+M_GPRIME_G_VAL 0 1 1 0.08
+
+#(l=1)
+M_GPRIME_G_VAL 1 0 0 -0.001
+M_GPRIME_G_VAL 1 0 1 0.001
+M_GPRIME_G_VAL 1 1 1 0.001
+TRANSFER_MOMENTS_END
+\endcode
+
+## Steady state simulations:
+
+The cross sections can be used in simulations solving the steady state
+Linear Boltzmann Equation of the form:
+
+\f[
+\vec{\Omega}_n \boldsymbol{\cdot} \vec{\nabla} \psi_{ng} +
+\sigma_{tg} \psi_{ng} =
+\sum_{\ell=0}^L \sum_{m=-\ell}^{+\ell} \frac{2\ell+1}{4\pi}
+    Y_{\ell m} (\vec{\Omega}_n)
+    \sum_{g'=0}^{G-1} \sigma_{s\ell,g'{\to}g} \phi_{\ell m,g'} +
+    q_{ext,ng} + q_{fission,ng}
+\f]
+
+The two most prominent items required here includes \f$ \sigma_{tg} \f$ and
+ \f$ \sigma_{s\ell,g'{\to}g} \f$. The latter is an entry in a structure we
+ call a generic transfer matrix for moment \f$ \ell \f$ with rows \f$g=0..G-1\f$
+ and columns \f$g'=0..G-1\f$. These two items are often the only items required
+ in a transport simulation.
+
+ In simulations with fission-sources, cross sections
+ support two formats, the simple combined cross sections without delayed
+ neutrons
+
+\f[
+q_{fission,ng} = \frac{\chi_g}{4\pi}
+ \sum_{g'=0}^{G-1} \nu_{g'} \sigma_{fg'} \phi_{00g'}
+\f]
+
+and those with delayed neutrons, the latter which are currently only used in the
+KEigen solver.
+
+## KEigen-value related items
+As stated before, the cross section file supports two formats, the simple combined cross sections without delayed
+neutrons which are depicted above and those with delayed neutrons shown below
+
+\f[
+q_{fission,ng} = \frac{\chi_g}{4\pi}
+ \sum_{g'=0}^{G-1} \nu_{prompt,g'} \sigma_{fg'} \phi_{00g'}
+ + \sum_{j=0}^{J-1} \frac{\chi_{delayed,jg}}{4\pi} \gamma_j
+    \sum_{g'=0}^{G-1} \nu_{delayed,g'} \sigma_{fg'} \phi_{00g'}
+\f].
+
+Codes may also choose to update precursor concentrations for which their decay
+constants, \f$ \lambda_j \f$ are required.
+
+## Keyword definitions
+
+- NUM_GROUPS num_groups Required. Specifies the number of groups for this
+  cross-section. Symbol \f$ G \f$.
+- NUM_MOMENTS num_moments Required. The number of transfer matrices to allocate
+  for this cross-section (whether used or not). Typically this number is one
+  greater than the scattering order (i.e., \f$ L+1 \f$)
+- NUM_PRECURSORS num_precursors Optional. Indicates how many precursors are used
+  in this cross section. Symbol \f$ J \f$
+- Optional key words per line:
+  - SIGMA_T_BEGIN. Optional. Starts a block that is terminated by a line SIGMA_T_END.
```

Comment:
how can sigma_t be optional?

Reply:
Because of the tranquility of simplicity. The most basic cross section file has two lines:

```
NUM_GROUPS 1
NUM_MOMENTS 0
```

That would supply a zero sigma_t.